tartiflette_plugin_scalars/duration.py

```diff
@@ -0,0 +1,70 @@
+from datetime import timedelta
+from typing import Union
+
+from tartiflette.constants import UNDEFINED_VALUE
+from tartiflette.language.ast import StringValueNode
+
+
+def _parse_duration(value: str) -> timedelta:
+    def get_kv(_arg: str) -> dict:
```

Comment:
can you extract this method ? it will improve readability IMHO.

Reply:
Did you mean moving it out of _parse_duration or just putting that logic where the method call is?

Reply:
i mean out of the `_parse_duration` method.

Reply:
@abusi Forgot to reply, this has been fixed